tests/ignite/handlers/test_ema_handler.py

```diff
@@ -224,8 +234,9 @@ def _test_ema_final_weight(model, device, ddp=False, interval=1):
     # engine will run 4 iterations
     engine.run(range(2), max_epochs=2)
 
-    ema_weight = ema_handler.ema_model.weight.data
-    model_weight = model.weight.data
+    # ema_model and model can be DP or DDP
+    ema_weight = _unwrap_model(ema_handler.ema_model).weight.data
+    model_weight = _unwrap_model(model).weight.data
     assert ema_weight.device == device
     assert model_weight.device == device
```

Comment:
```suggestion
    assert ema_weight.device.type == device.type
    assert model_weight.device == ema_weight.device
```

Reply:
The problem is that in the horovod test, the processes will be spawn by the horovod executor. Thus, if we pass `device = "cuda"` to the function, `model = idist.auto_model(model)` will move the `model` to `idist.device()`, which might be `f"cuda:{idist.local_rank()}"`, however `device` is `torch.device("cuda")`. In order to fix this, I change the signature of `_test_ema_final_weight` to `_test_ema_final_weight(model, device=None, ddp=False, interval=1)`. If `device=None`, we call `idsti.device()` to get the current device. Then, we explicitly pass `None` to this function in the horovod executor. I have run the tests on my server with single node, and 22 passed, 6 skipped.

Reply:
Yes, your solution will work as well and looks good to me. Thanks !